automata/base/animation.py

```diff
@@ -0,0 +1,254 @@
+from __future__ import annotations
+
+from collections.abc import Iterable
+from dataclasses import dataclass
+from functools import partial
+from typing import Optional, TypeVar
+
+import manim
+import pygraphviz as pgv
+
+_POINTS_IN_INCH = 72
+
+
+class Animate:
+    """The `Animate` class is the config class to set the behavior of animations."""
+
+    DEFAULT_COLOR = manim.WHITE
+    HIGHLIGHT_COLOR = manim.RED
+    M = TypeVar("M", bound=manim.Mobject)
+
+    @classmethod
+    def default_init(cls, mobject_class: type[M]) -> type[M]:
+        return partial(mobject_class, color=cls.DEFAULT_COLOR)  # type: ignore
+
+    @classmethod
+    def to_default_color(cls, mobject: manim.VMobject) -> manim.ApplyMethod:
+        """
+        Parameters
+        ----------
+        mobject : VMobject
+            The mobject to change to `DEFAULT_COLOR`, or cancel the highlight.
+
+        Returns
+        -------
+        ApplyMethod
+            The animation for the `Scene` object to `play` to change the mobject to
+            `DEFAULT_COLOR`.
+        """
+        return manim.FadeToColor(mobject, cls.DEFAULT_COLOR)
+
+    @classmethod
+    def highlight(cls, mobject: manim.VMobject) -> manim.ApplyMethod:
+        """
+        Parameters
+        ----------
+        mobject : VMobject
+            The mobject to set the `HIGHLIGHT_COLOR`.
+
+        Returns
+        -------
+        ApplyMethod
+            The animation for the `Scene` object to `play` to highlight the mobject.
+        """
+        return manim.FadeToColor(mobject, cls.HIGHLIGHT_COLOR)
+
+
+@dataclass(eq=False)
```

Comment:
Why do you need to use the dataclass decorator if you have an init function?

Reply:
You do not need the dataclass decorator to do this: https://stackoverflow.com/questions/53897054/python-class-members-type-hinting

The dataclass decorator is used to avoid code duplication by automatically generating some methods based on member variables. Unless you are using those generated methods, then you should not be using this decorator.